src/liquid/components/ld-stepper/ld-step/ld-step.tsx

```diff
@@ -0,0 +1,164 @@
+  tag: 'ld-step',
+  styleUrl: 'ld-step.css',
+  shadow: true,
+})
+export class LdStep implements InnerFocusable {
+  @Element() el: HTMLLdStepElement
+  private focusableElement: HTMLButtonElement | HTMLAnchorElement
+
+  /** Switch colors for brand background */
+  @Prop() brandColor = false
+  /** Step is the current step */
+  @Prop() current = false
+  /** Description text to display below the step name (vertical mode only) */
+  @Prop() description: string
+  /** Step is not clickable */
+  @Prop() disabled = false
+  /** Step is done */
+  @Prop() done = false
+  /** Link to the step (makes the step an anchor instead of a button) */
+  @Prop() href?: string
+  /** Permanently show a custom icon inside the dot */
+  @Prop() icon?: HTMLLdIconElement['name']
+  /** Label for current step (scree-reader only) */
+  @Prop() labelCurrent = 'Current'
+  /** Label for step that is done (scree-reader only) */
+  @Prop() labelDone = 'Done'
+  /** Label for step that is optional (scree-reader only) */
+  @Prop() labelOptional = 'Optional'
+  /** Label for step that was skipped (scree-reader only) */
+  @Prop() labelSkipped = 'Skipped'
+  /** Additional hint in label for step that is done and was optional (scree-reader only) */
+  @Prop() labelWasOptional = 'was optional'
+  /** Indicates that the next step is not active */
+  @Prop() lastActive = false
+  /** Tab index of the step */
+  @Prop() ldTabindex: number | undefined
+  /** Step can be processed next */
+  @Prop() next = false
+  /** Step may be skipped */
+  @Prop() optional = false
+  /** Step size */
+  @Prop() size?: 'sm' | 'lg'
+  /** Step was skipped */
+  @Prop() skipped = false
+  /** Vertical layout */
+  @Prop() vertical = false
+
+  @Event() ldstepselected: EventEmitter<SelectedDetail>
+
+  /** Sets focus on the step */
+  @Method()
+  async focusInner() {
+    this.focusableElement?.focus()
+  }
+
+  handleClick = () => {
+    this.ldstepselected.emit({
+      index: Array.from(this.el.parentElement.children).findIndex(
+        (child) => child === this.el
+      ),
+      label: this.el.textContent,
+    })
+  }
+
+  componentDidLoad() {
+    if (this.current) {
+      this.handleClick()
+    }
+  }
+
+  render() {
+    const Component = this.href ? 'a' : 'button'
+
+    return (
+      <Host>
+        <li
+          class={getClassNames([
+            'ld-step',
+            this.brandColor && 'ld-step--brand-color',
+            this.current && 'ld-step--current',
+            this.done && 'ld-step--done',
+            this.icon && 'ld-step--custom-icon',
+            (this.done || this.icon) && 'ld-step--with-icon',
+            this.lastActive && 'ld-step--last-active',
+            this.next && 'ld-step--next',
+            this.optional && 'ld-step--optional',
+            this.size && `ld-step--${this.size}`,
+            this.skipped && 'ld-step--skipped',
+            this.vertical && 'ld-step--vertical',
+          ])}
+          part="li"
+          role="listitem"
```

Comment:
`role` shouldn't be necessary here as the `li` has the role implicitly.

Reply:
I've read [here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listitem_role#best_practices) that if I use `list-style: none`, the role is required to make it a list item again. Leaving as is.